src/ert/job_queue/realization_state.py

```diff
@@ -135,24 +136,27 @@ def on_enter_EXIT(self) -> None:
                 f"Realization: {self.realization.run_arg.iens} "
                 f"failed after reaching max submit ({self._max_submit}):"
             )
-            exit_file_path = (
+            log_info_from_exit_file(
                 Path(self.realization.run_arg.runpath) / self.realization.exit_file
             )
-            if exit_file_path.exists():
-                exit_file = etree.parse(exit_file_path)
-                failed_job = exit_file.find("job").text
-                error_reason = exit_file.find("reason").text
-                stderr_capture = exit_file.find("stderr").text
-                stderr_file = exit_file.find("stderr_file").text
-                logger.error(
-                    f"job {failed_job} failed with: '{error_reason}'\n"
-                    f"\tstderr file: '{stderr_file}',\n"
-                    f"\tits contents:{stderr_capture}"
-                )
             self.invalidate()
 
     def on_enter_DONE(self) -> None:
         asyncio.create_task(self.jobqueue.run_done_callback(self))
 
     def on_enter_DO_KILL(self) -> None:
         asyncio.create_task(self.jobqueue.driver.kill(self))
+
+
+def log_info_from_exit_file(exit_file_path: Path) -> None:
+    if not exit_file_path.exists():
+        return
+    exit_file = etree.parse(exit_file_path)
+    filecontents: Dict[str, str] = defaultdict(str)
```

Comment:
It's nice to move this out to a function, but unpacking from the etree to a dict, then from the dict to the f-strings feels a little roundabout. It was maybe a bit repetitive before, but I think it was clearer (though I tripped a little on the order).

Maybe you could use something like `f"...".format(**filecontents)` to save the unpacking? And even just use a `list` instead of a `dict` and unpack like `*filecontents` into unlabeled `{}` pairs?

IDK, not a bit deal either way, just a thought.

Reply:
Added an alternative, check if it looks better (I find it hard to decide..)